sdk/src/main/kotlin/io/rover/sdk/ui/blocks/poll/VotingInteractor.kt

```diff
@@ -117,26 +119,30 @@ internal class VotingInteractor(
     private fun votingResultsUpdate(optionIds: List<String>) {
         refreshResultsHandler.removeCallbacksAndMessages(null)
 
-        fetchVotingResults(optionIds).observeOn(mainScheduler).first().subscribe(
-            { fetchedOptionResults ->
-                val state = currentState
-                if (state is VotingState.RefreshingResults && fetchedOptionResults.results.isNotEmpty()) {
-                    currentState = state.transitionToRefreshingResults(
-                        changeVotesToPercentages(fetchedOptionResults),
-                        shouldTransition = true,
-                        shouldAnimate = true
-                    )
-                }
-            },
-            { if (currentState is VotingState.RefreshingResults) refreshResults() },
-            { subscription ->
-                if (cancelled) {
-                    subscription.cancel()
-                } else {
-                    currentUpdateSubscription = subscription
-                    subscriptions.add(subscription)
-                }
-            })
+        if (pollVisible) {
+            fetchVotingResults(optionIds).observeOn(mainScheduler).first().subscribe(
+                { fetchedOptionResults ->
+                    val state = currentState
+                    if (state is VotingState.RefreshingResults && fetchedOptionResults.results.isNotEmpty()) {
+                        currentState = state.transitionToRefreshingResults(
+                            changeVotesToPercentages(fetchedOptionResults),
+                            shouldTransition = true,
+                            shouldAnimate = true
+                        )
+                    }
+                },
+                { if (currentState is VotingState.RefreshingResults) refreshResults() },
+                { subscription ->
+                    if (cancelled) {
+                        subscription.cancel()
+                    } else {
+                        currentUpdateSubscription = subscription
+                        subscriptions.add(subscription)
+                    }
+                })
+        } else {
+            currentState = currentState
+        }
```

Comment:
Rather than promoting the visibility concern into the Interactor, would it be possible to instead clean up/shut down the timer/refreshing behaviour on the View unsubscribing from the Publisher?